### PR TITLE
Update anchoring guide

### PR DESCRIPTION
**Pull request by alekseysidorov**

### CODE REVIEW

File: src/advanced/bitcoin-anchoring.md [hunk 1]
```diff
@@ -263,59 +273,9 @@ The funding UTXO should get enough confirmations before it can be used.
...
-transaction should be committed to the Bitcoin blockchain **before** the updated
-list of validators comes into force. Thus the anchoring process is suspended.
-
-- After the latest anchoring transactions is committed to the Bitcoin
-  blockchain, a transitional anchoring transaction proposal is generated. That
-  transaction moves money to the new anchoring address.
-- As the anchoring chain has already been moved to the new anchoring address,
-  Exonum nodes wait until the new validators set comes into force; after that
-  the anchoring process resumes at the new address.
+Fresh and full maintenance guide you can alway find [here][maintenance-guide].
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Grammar: I'd rephrase as "The maintenance guide is here" (or "...can be found here").

File: src/advanced/bitcoin-anchoring.md [hunk 2]
```diff
@@ -334,10 +294,9 @@ The service provides the following public API endpoints:
 - [Get the actual anchoring address](#actual-address)
 - [Get the following anchoring address](#following-address)
 - [Get an anchoring transaction](#get-an-anchoring-transaction)
-- [Get a proof for an Exonum block](#cryptographic-proof-for-an-exonum-block)
 
 All REST endpoints share the same base path, denoted **{api_prefix}**,
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: We've generally moved to describing HTTP API in the corresponding crate, because it's easier to keep it updated and test it this way. Do I understand correctly that testing the anchoring service endpoints that way would require non-trivial effort?

File: src/advanced/bitcoin-anchoring.md [hunk 3]
```diff
@@ -168,24 +168,25 @@ anchoring transaction only if the previous anchoring chain failed
...
+  and sends all such anchoring transactions to Bitcoin. Therefore, even if some
+  committed anchoring transactions are lost from the network due to a fork,
+  the handler will send them to Bitcoin once again.
 
 ## Setup and Configuration
 
 Anchoring requires additional
 [global and local configuration parameters](../architecture/configuration.md) to
-be set.
+be set. [Here][newbie-guide] you can find setup and deploy guide for newbies.
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Why isn't this called simply a setup guide? As far as I see, there are no other setup guides 🙃

File: src/advanced/bitcoin-anchoring.md [hunk 4]
```diff
@@ -214,13 +215,22 @@ secured.
...
 #### Synchronization with Bitcoin Blockchain
 
-A separate [method][handler] performs regular synchronization of the list of
-Exonum anchoring transactions with those committed to the Bitcoin Blockchain. It
-takes the latest
-anchoring transaction and checks whether it is present in Bitcoin. If not, the
-handler checks the previous anchoring transactions one by one in the same manner
-until it finds the last anchoring transaction committed to Bitcoin Blockchain.
-The handler then pushes all the uncommitted anchoring transactions to Bitcoin.
+A separate [btc_anchoring_sync][btc_anchoring_sync] utility periodically performs
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Nit: `btc_anchoring_sync` should be code-formatted, although I'd remove the name altogether and linked from "A separate utility" instead.

File: src/advanced/bitcoin-anchoring.md [hunk 5]
```diff
@@ -168,24 +168,25 @@ anchoring transaction only if the previous anchoring chain failed
   anchoring transaction and the hash of the corresponding Exonum block. The
   first anchoring transaction in the chain uses the UTXO of the funding
   transaction
-- Every validator creates an Exonum transaction containing a new proposed
-  anchoring transaction and its vote for said transaction. The validators commit
-  these transactions to the Exonum blockchain
+- Every validator node has an external sync [util][btc_anchoring_sync] that
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Nit: Here and in some other places, you use "validator" - although, if I understand correctly, anchoring is no longer performed necessarily by the blockchain validators. I'd use introduce a term like "anchoring node" and use it instead.

File: src/advanced/bitcoin-anchoring.md [hunk 6]
```diff
@@ -214,13 +215,22 @@ secured.
...
-Exonum anchoring transactions with those committed to the Bitcoin Blockchain. It
-takes the latest
-anchoring transaction and checks whether it is present in Bitcoin. If not, the
-handler checks the previous anchoring transactions one by one in the same manner
-until it finds the last anchoring transaction committed to Bitcoin Blockchain.
-The handler then pushes all the uncommitted anchoring transactions to Bitcoin.
+A separate [btc_anchoring_sync][btc_anchoring_sync] utility periodically performs
+two actions:
+
+- Creation of a signature for a new anchoring transaction. It takes the actual
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Nit / grammar: It -> The utility

File: src/advanced/bitcoin-anchoring.md [hunk 7]
```diff
@@ -214,13 +215,22 @@ secured.
...
+  anchoring transaction proposal by the node private API, signs this proposal by
+  the corresponding Bitcoin key and sends this signature back to the validator
+  node. Validator node creates a new vote transaction from this signature and
+  broadcasts it to the other nodes.
+
+- Synchronization of the list of Exonum anchoring transactions with those committed
+  to the Bitcoin Blockchain. It takes the latest anchoring transaction and checks
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Nit / grammar: It -> The utility

File: src/advanced/bitcoin-anchoring.md [hunk 8]
```diff
@@ -346,7 +305,7 @@ equal to `/api/services/btc_anchoring`.
 ### Actual Address
```
Comment (slowli) [inline src/advanced/bitcoin-anchoring.md]: Do you plan to describe private endpoints?

File: src/advanced/bitcoin-anchoring.md (hunk 3 above)
Comment (popzxc) [reply src/advanced/bitcoin-anchoring.md]: It's because of me. When I wrote it, there was a primary (outdated though) setup guide which had much less details. So I called it a "guide for newbies", since it was detailed to the last command. But later it became the primary thing, but was not renamed...

File: src/advanced/bitcoin-anchoring.md (hunk 2 above)
Comment (alekseysidorov) [reply src/advanced/bitcoin-anchoring.md]: Anchoring crate already has API endpoints description, but not such verbose as here.

File: src/advanced/bitcoin-anchoring.md (hunk 8 above)
Comment (alekseysidorov) [reply src/advanced/bitcoin-anchoring.md]: No, I don't think because this is an internal implementation details and the end user should not know about such details.

File: src/advanced/bitcoin-anchoring.md [hunk 9]
```diff
@@ -10,6 +10,11 @@ Even in the case of validators collusion, transaction history cannot be
 falsified. The discrepancy between the actual Exonum blockchain state and the
 one written to the Bitcoin blockchain will be found instantly.
 
+This document describes the **anchoring service operable with Exonum v0.10+**.
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]: 0.10?

File: src/advanced/bitcoin-anchoring.md [hunk 10]
```diff
@@ -163,24 +168,25 @@ anchoring transaction only if the previous anchoring chain failed
   anchoring transaction and the hash of the corresponding Exonum block. The
   first anchoring transaction in the chain uses the UTXO of the funding
   transaction
-- Every validator creates an Exonum transaction containing a new proposed
-  anchoring transaction and its vote for said transaction. The validators commit
-  these transactions to the Exonum blockchain
+- Every validator has an external sync [util][btc_anchoring_sync] that
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]: I'd use "synchronization utility" instead of "sync util".

File: src/advanced/bitcoin-anchoring.md [hunk 11]
```diff
@@ -209,13 +215,22 @@ secured.
...
-The handler then pushes all the uncommitted anchoring transactions to Bitcoin.
+A separate [utility][btc_anchoring_sync] periodically performs
+two actions:
+
+- Creation of a signature for a new anchoring transaction. The utility takes the
+  actual anchoring transaction proposal by the node private API, signs this
+  proposal by the corresponding Bitcoin key and sends this signature back to the
+  validator node. Validator node creates a new vote transaction from this signature
+  and broadcasts it to the other nodes.
+
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]: Isn't empty line between list elements breaks the list?

File: src/advanced/bitcoin-anchoring.md [hunk 12]
```diff
@@ -209,13 +215,22 @@ secured.
...
 
-A separate [method][handler] performs regular synchronization of the list of
-Exonum anchoring transactions with those committed to the Bitcoin Blockchain. It
-takes the latest
-anchoring transaction and checks whether it is present in Bitcoin. If not, the
-handler checks the previous anchoring transactions one by one in the same manner
-until it finds the last anchoring transaction committed to Bitcoin Blockchain.
-The handler then pushes all the uncommitted anchoring transactions to Bitcoin.
+A separate [utility][btc_anchoring_sync] periodically performs
+two actions:
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]:
```suggestion
the following actions:
```

File: src/advanced/bitcoin-anchoring.md [hunk 13]
```diff
@@ -209,13 +215,22 @@ secured.
...
+
+- Creation of a signature for a new anchoring transaction. The utility takes the
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]:
```suggestion
1. Creates a signature for a new anchoring transaction. The utility takes the
```

File: src/advanced/bitcoin-anchoring.md [hunk 14]
```diff
@@ -209,13 +215,22 @@ secured.
...
+  actual anchoring transaction proposal by the node private API, signs this
+  proposal by the corresponding Bitcoin key and sends this signature back to the
+  validator node. Validator node creates a new vote transaction from this signature
+  and broadcasts it to the other nodes.
+
+- Synchronization of the list of Exonum anchoring transactions with those committed
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]:
```suggestion
2. Synchronizes the list of Exonum anchoring transactions with those committed
```

File: src/advanced/bitcoin-anchoring.md (hunk 11 above)
Comment (alekseysidorov) [reply src/advanced/bitcoin-anchoring.md]: No, list looks good in preview.

File: src/advanced/bitcoin-anchoring.md [hunk 15]
```diff
@@ -198,30 +204,39 @@ The following settings need to be specified to access the bitcoind node:
...
+the following actions:
+
+1. Creates a signature for a new anchoring transaction. The utility takes the
+  actual anchoring transaction proposal by the node private API, signs this
+  proposal by the corresponding Bitcoin key and sends this signature back to the
+  anchoring node. Anchoring node creates a new vote transaction from this signature
+  and broadcasts it to the other nodes.
+
+2. Synchronizes the list of Exonum anchoring transactions with those committed
+  to the Bitcoin Blockchain. The utility the latest anchoring transaction and checks
```
Comment (popzxc) [inline src/advanced/bitcoin-anchoring.md]: > The utility the latest anchoring transaction

?
